components/script/dom/htmlimageelement.rs

```diff
@@ -429,6 +473,40 @@ impl VirtualMethods for HTMLImageElement {
             _ => self.super_type().unwrap().parse_plain_attribute(name, value),
         }
     }
+
+    fn handle_event(&self, event: &Event) {
+          //let node = self.upcast::<Node>();
+       if (event.type_() == atom!("click")) {
+           let area_elements = self.areas();
+           let elements;
+           match area_elements {
+           Some(x) => elements = x,
+           None => return,
+           };
+           //let elements = area_elements.unwrap();
+
+           // Fetch click coordinates
+           let mouse_event = event.downcast::<MouseEvent>().unwrap();
```

Comment:
We cannot rely on the type working this way. Please use an if let and return if the event is not a mouse event.